src/middlewared/middlewared/plugins/system_/coredump.py

```diff
@@ -1,33 +1,30 @@
-# -*- coding=utf-8 -*-
-import logging
+from os.path import exists
+from systemd import journal
 
 from middlewared.service import private, Service
-from middlewared.utils import run
-
-logger = logging.getLogger(__name__)
 
 
 class SystemService(Service):
+
     @private
-    async def coredumps(self):
+    def coredumps(self):
         coredumps = []
-        coredumpctl = await run("coredumpctl", "list", "--no-pager", encoding="utf-8", errors="ignore", check=False)
-        if coredumpctl.returncode != 0:
-            return []
 
-        lines = coredumpctl.stdout.splitlines()
-        header = lines.pop(0)
-        exe_pos = header.find("EXE")
-        for line in lines:
-            exe = line[exe_pos:]
-            time, pid, uid, gid, sig, corefile = line[:exe_pos].rsplit(maxsplit=5)
-            coredumps.append({
-                "time": time,
-                "pid": int(pid),
-                "uid": int(uid),
-                "gid": int(gid),
-                "sig": int(sig),
-                "corefile": corefile,
-                "exe": exe,
-            })
+        try:
+            with journal.Reader() as reader:
+                reader.add_match(CODE_FUNC='submit_coredump')
+                for core in reader:
+                    coredumps.append({
+                        'time': core['COREDUMP_TIMESTAMP'].strftime('%c'),
+                        'pid': core['COREDUMP_PID'],
+                        'uid': core['COREDUMP_UID'],
+                        'gid': core['COREDUMP_GID'],
+                        'unit': core['COREDUMP_UNIT'],
+                        'sig': core['COREDUMP_SIGNAL'],
+                        'corefile': 'present' if exists(core['COREDUMP_FILENAME']) else 'missing',
```

Comment:
`coredumpctl` can show `none` here if corefile failed to generate rather than having it simply be missing. Have we checked whether this edge cases causes `core['COREDUMP_FILENAME']` to be `None` rather than a string? If it's None type then `exists(core['COREDUMP_FILENAME'])` will raise a TypeError.

Reply:
Simple way to check is to set a 1 MiB quota on the <sysdataset>/cores and then kill -6 an unwitting victim.